src/StackExchange.Redis/StreamAutoClaimResult.cs

```diff
@@ -0,0 +1,32 @@
+﻿using System;
+
+namespace StackExchange.Redis
+{
+    /// <summary>
+    /// The result of the XAUTOCLAIM command.
+    /// </summary>
+    public readonly struct StreamAutoClaimResult
+    {
+        internal StreamAutoClaimResult(RedisValue nextStartId, StreamEntry[] claimedEntries, RedisValue[] deletedIds)
+        {
+            NextStartId = nextStartId;
+            ClaimedEntries = claimedEntries ?? Array.Empty<StreamEntry>();
+            DeletedIds = deletedIds ?? Array.Empty<RedisValue>();
```

Comment:
@NickCraver understood for claimedEntries, but deletedIds could be null internally when running 6.2 because the call to GetItemsAsValues would only happen for >= 7.0.

Reply:
Yes, but that's null coalesced in the constructor call so it's never null here unless I'm missing something. The args to this method are internal and non-nullable :) That combo forces the upstream to choose what to pass (instead of null) and you're already taking care of it. Though we need to split this into 2 methods anyhow, your upstream usage to call this is correct so can remove the coalesces here.

Reply:
@NickCraver yes, you're right. Should be fine since I'm coalescing when calling the constructor. My mistake.

Reply:
@ttingen no worries! Sorry some comments are terse, trying to fit a lot in before mostly away next week and was a crammed day <3